src/rgw/rgw_sal_dbstore_config.cc

```diff
@@ -0,0 +1,880 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include "rgw_sal_dbstore_config.h"
+#include "common/async/yield_context.h"
+#include "common/dout.h"
+#include "common/random_string.h"
+#include "rgw_zone.h"
+
+namespace rgw::sal {
+
+constexpr std::string_view create_table_realms =
+R"(CREATE TABLE IF NOT EXISTS Realms (
+ID TEXT PRIMARY KEY NOT NULL,
+Name TEXT UNIQUE NOT NULL,
+CurrentPeriod TEXT REFERENCES Periods (ID),
+Epoch INTEGER DEFAULT 0,
+VersionNumber INTEGER,
+VersionTag TEXT
```

Comment:
@soumyakoduri @mattbenjamin i designed a db schema and queries for the storage of realm/period/zonegroup/zone objects, but haven't integrated it with anything under rgw/rgw/store/dbstore yet

if you guys have a chance, could you please look over the SQL here?

Reply:
sure. Thanks Casey!